start code gen implementation

jlvivero opened the pull request:
It's still incomplete

jlvivero committed:
start code gen implementation

It's still incomplete

jlvivero committed:
implement basic code generation functions

missing jumps, and missing the actual stdout implemenation of the code generation

jlvivero commented:
I have a question regarding how labels work @A-deLuna , and how to implement the codegen for jumps

jlvivero committed:
Add jump and label functions

needs stdout conditions for printing either asm or binary code
still needs testing

alexispoo reviewed (dismissed):
New methods that holds the variable that will be used for writing which is perfect, method for and if seems fine for me,notit also seems good. Added 4 methods as well to code_gen.h in order to get them working, for my point of view the code it's good to go.

jlvivero committed:
Implement code generation methods

Added a flag for generating asm
to generate assembly code, add asm as argument when running code
this needs testing

jlvivero committed:
merge master into code-gen

jlvivero committed:
Fix Code generation bug

It works now, add flag asm for out on asm, bin (for convenience) for binary

jlvivero commented:
has a bug with if/for/while, I believe it's a label bug

A-deLuna committed:
codegen works!
